go.mod

```diff
@@ -26,47 +27,44 @@ require (
 	github.com/spf13/cobra v1.0.0
 	github.com/stretchr/testify v1.5.1
 	golang.org/x/net v0.0.0-20200904194848-62affa334b73
-	google.golang.org/grpc v1.31.1
+	google.golang.org/grpc v1.33.1
 	google.golang.org/protobuf v1.25.0
 	istio.io/api v0.0.0-20200715212100-dbf5277541ef
 	istio.io/client-go v0.0.0-20201005161859-d8818315d678
 	istio.io/gogo-genproto v0.0.0-20191029161641-f7d19ec0141d // indirect
-	k8s.io/api v0.18.8
-	k8s.io/apiextensions-apiserver v0.18.8
-	k8s.io/apimachinery v0.18.8
+	k8s.io/api v0.19.0
+	k8s.io/apiextensions-apiserver v0.19.0 // indirect
+	k8s.io/apimachinery v0.19.0
+	k8s.io/apiserver v0.19.0
 	k8s.io/client-go v11.0.1-0.20190805182717-6502b5e7b1b5+incompatible
 	k8s.io/klog v1.0.0
-	k8s.io/kube-openapi v0.0.0-20200410145947-bcb3869e6f29
-	knative.dev/pkg v0.0.0-20200922164940-4bf40ad82aab
-	knative.dev/serving v0.18.0
-	sigs.k8s.io/controller-runtime v0.6.3
+	k8s.io/kube-openapi v0.0.0-20200805222855-6aeccd4b50c6
+	knative.dev/pkg v0.0.0-20201029081435-565516e2247d
+	knative.dev/serving v0.18.1
+	sigs.k8s.io/controller-runtime v0.6.2
+	sigs.k8s.io/controller-tools v0.3.0
 )
 
 replace (
-	k8s.io/api => k8s.io/api v0.18.8
-	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.18.8
-	k8s.io/apimachinery => k8s.io/apimachinery v0.18.8
-	k8s.io/apiserver => k8s.io/apiserver v0.18.8
-	k8s.io/cli-runtime => k8s.io/cli-runtime v0.18.8
-	k8s.io/client-go => k8s.io/client-go v0.18.8
-	k8s.io/cloud-provider => k8s.io/cloud-provider v0.18.8
-	k8s.io/cluster-bootstrap => k8s.io/cluster-bootstrap v0.18.8
-	k8s.io/code-generator => k8s.io/code-generator v0.18.8
-	k8s.io/component-base => k8s.io/component-base v0.18.8
-	k8s.io/cri-api => k8s.io/cri-api v0.18.8
-	k8s.io/csi-translation-lib => k8s.io/csi-translation-lib v0.18.8
-	k8s.io/kube-aggregator => k8s.io/kube-aggregator v0.18.8
-	k8s.io/kube-controller-manager => k8s.io/kube-controller-manager v0.18.8
-	k8s.io/kube-proxy => k8s.io/kube-proxy v0.18.8
-	k8s.io/kube-scheduler => k8s.io/kube-scheduler v0.18.8
-	k8s.io/kubectl => k8s.io/kubectl v0.18.8
-	k8s.io/kubelet => k8s.io/kubelet v0.18.8
-	k8s.io/legacy-cloud-providers => k8s.io/legacy-cloud-providers v0.18.8
-	k8s.io/metrics => k8s.io/metrics v0.18.8
-	k8s.io/sample-apiserver => k8s.io/sample-apiserver v0.18.8
-	k8s.io/utils => k8s.io/utils v0.0.0-20200324210504-a9aa75ae1b89
-	k8s.io/test-infra => k8s.io/test-infra v0.0.0-20200803112140-d8aa4e063646
-	k8s.io/klog => k8s.io/klog v1.0.0
-	// based on https://github.com/openshift/cluster-ingress-operator/pull/429/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6R34
-	sigs.k8s.io/controller-runtime => github.com/zchee/sigs.k8s-controller-runtime v0.6.1-0.20200623114430-46812d3a0a50
+	k8s.io/api => k8s.io/api v0.19.0
```

Comment:
@ryandawsonuk Did you try 0.18.6 according to last comment https://github.com/kubernetes-sigs/controller-runtime/issues/1033 conroller-runtime v0.6.x should work with that?

Reply:
Weirdly trying that leads me to https://github.com/kubernetes/client-go/issues/808. I tried to work around it with replace to bring down the gnostic version but then I get the lowercase version of the error - https://github.com/kubernetes/client-go/issues/741. It seems that [1.19 does indeed fix that](https://github.com/kubernetes/client-go/issues/741#issuecomment-606883355) as this compiles with 1.19. Not looked into e2e test failures though.